packages/app/src/cli/services/app_logs/fetch-app-logs.ts

```diff
@@ -0,0 +1,79 @@
+import {writeAppLogsToFile} from './write-app-logs.js'
+import {partnersFqdn} from '@shopify/cli-kit/node/context/fqdn'
+import {Writable} from 'stream'
+
+const generateFetchAppLogUrl = async (cursor?: string) => {
+  const fqdn = await partnersFqdn()
+  const url = `https://${fqdn}//app_logs/poll`
+  return url + (cursor ? `?cursor=${cursor}` : '')
+}
+
+export interface AppEventData {
+  shop_id: number
+  api_client_id: number
+  payload: string
+  event_type: string
+  cursor: string
+  status: 'success' | 'failure'
+  log_timestamp: string
+}
+
+export const fetchAppLogs = async ({
+  stdout,
+  appLogsFetchInput: {jwtToken, cursor},
+  apiKey,
+}: {
+  stdout: Writable
+  appLogsFetchInput: {jwtToken: string; cursor?: string}
+  apiKey: string
+}) => {
+  const url = await generateFetchAppLogUrl(cursor)
+  const response = await fetch(url, {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${jwtToken}`,
+    },
+  })
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`)
+  }
+
+  const data = (await response.json()) as {
+    app_logs?: AppEventData[]
+    cursor?: string
+    errors?: string[]
+  }
+
+  if (data?.errors) {
+    data?.errors?.forEach((error) => {
+      stdout.write(`${error}\n`)
+    })
+  } else if (data.app_logs) {
+    const {app_logs: appLogs} = data
+
+    await Promise.all(
+      appLogs.map(async (log) => {
+        await writeAppLogsToFile({
+          appLog: log,
+          apiKey,
+          stdout,
+        })
+      }),
+    )
+  }
+
+  const cursorFromResponse = data?.cursor
+
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  setTimeout(async () => {
+    await fetchAppLogs({
+      stdout,
+      appLogsFetchInput: {
+        jwtToken,
+        cursor: cursorFromResponse,
+      },
+      apiKey,
+    })
+  }, 1000)
```

Comment:
I think this should be set as a constant

Reply:
@DuncanUszkay1 We figured that we needed 400ms to meet our SLO right?

I thought we also discussed providing some sort of server-side control over this (a header returned during the first poll?) so that we can adjust if needed. Once a CLI version is out, we can't change a hardcoded value.

Reply:
I think it's fine to ship this (at 450ms) behind an internal beta flag for this PR- ultimately users can modify their client to poll at any rate, so it's fine if it appears in the code. The actual guard against over-polling is serverside, and it's an IP throttle.

As a follow-up, we can add in the serverside interval.